components/collapsible-panel/collapsible-panel.js

```diff
@@ -0,0 +1,344 @@
+				margin-inline: 0.9rem;
+			}
+			.d2l-collapsible-panel-header-primary {
+				align-items: center;
+				display: flex;
+				justify-content: space-between;
+			}
+			.d2l-collapsible-panel-body {
+				padding: 0.9rem;
+			}
+			.d2l-collapsible-panel-content,
+			.d2l-collapsible-panel-summary {
+				margin-top: 0.6rem;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.headingLevel = undefined;
+		this.expanded = false;
+		this._hasSummary = false;
+		this.type = 'default';
+		this.fullWidth = false;
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this._intersectionObserver) this._intersectionObserver.disconnect();
+	}
+
+	firstUpdated() {
+		if (typeof(IntersectionObserver) === 'function') {
+			this._intersectionObserver = new IntersectionObserver(entries => {
+				if (!this.expanded) return;
+
+				const entry = entries[0];
+				this._scrolled = !entry.isIntersecting;
+			});
+			this._intersectionObserver.observe(this.shadowRoot.querySelector('.d2l-collapsible-panel-top-sentinel'));
+		}
+	}
+
+	render() {
+		return html`
+			<div class="d2l-collapsible-panel" @click="${this._handlePanelClick}">
+				<div class="d2l-collapsible-panel-top-sentinel"></div>
+				${this._renderHeader()}
+					<div class="d2l-collapsible-panel-body">
+						<div
+							class="d2l-collapsible-panel-content"
+							?hidden="${!this.expanded}"
+							.animate="${ this.expanded ? show() : hide()}"
+						>
+							<slot name="content"></slot>
+						</div>
+						<div
+							class="d2l-collapsible-panel-summary"
+							?hidden="${this.expanded}"
+							.animate="${ !this.expanded ? show() : hide()}"
+						>
+							<slot name="summary" @slotchange="${this._handleSummarySlotChange}"></slot>
+						</div>
+					</div>
+			</div>
+		`;
+	}
+
+	_handleActionsClick(e) {
+		const actions = this.shadowRoot.querySelector('.d2l-collapsible-panel-header-actions');
+		if (e.target !== actions) {
+			e.stopPropagation();
+		}
+	}
+
+	_handleHeaderClick(e) {
+		if (this.expanded) {
+			this._toggleExpand();
+			e.stopPropagation();
+		}
+	}
+
+	_handleHeaderSecondaryClick(e) {
+		const header = this.shadowRoot.querySelector('.d2l-collapsible-panel-header-secondary');
+		if (e.target !== header) {
+			e.stopPropagation();
+		}
+	}
+
+	_handlePanelClick(e) {
+		if (!this.expanded) {
+			this._toggleExpand();
+			e.stopPropagation();
+		}
+	}
+
+	_handleSummarySlotChange() {
+		this._hasSummary = true;
+	}
+
+	_onKeyDown(e) {
+		if (e.target.classList.contains('d2l-collapsible-panel-header') && (e.keyCode === keyCodes.ENTER || e.keyCode === keyCodes.SPACE)) {
+			this._toggleExpand();
+			e.preventDefault();
+		}
+	}
+
+	_renderHeader() {
+		const icon = 'tier1:arrow-expand';
+		// const icon = `tier1:arrow-${this.expanded ? 'collapse' : 'expand'}`;
+
+		return html`
+			<div class="d2l-collapsible-panel-header" @click="${this._handleHeaderClick}" @keydown="${this._onKeyDown}" tabindex="0">
```

Comment:
I think we'll want to be careful of a11y here. This `div` would be focusable but would not have any semantic meaning. But having a `button` element that contains other focusable and interactive elements can be problematic I think. 

So as mentioned in discussion, I may be inclined to have an offscreen button that receives focus and the label. Then you could just use a `click` handler on it... no need for special `keydown` handler, you can manage the focus order, and the focusable elements aren't nested. You'd have to listen for `focus`/`blur` to in turn render a class in this component for the focus state styling (blue ring).